core/src/main/java/org/apache/carbondata/core/cache/dictionary/Dictionary.java

```diff
@@ -59,6 +59,17 @@
   String getDictionaryValueForKey(int surrogateKey);
 
   /**
+   * This method will find and return the dictionary value for a given surrogate key in bytes.
+   * Applicable scenarios:
+   * 1. Query final result preparation : While convert the final result which will
+   * be surrogate key back to original dictionary values this method will be used
```

Comment:
It is added because `getDictionaryValueForKey` gets the bytes from dictionary and convert to String, but in our decoder we again convert to bytes. So I added new method to return bytes directly with out converting to String to avoid multiple conversions.